pkg/kratos/service.go

```diff
@@ -387,6 +391,74 @@ func (s *Service) UpdateRecoveryFlow(
 	return returnToResp, resp.Cookies(), nil
 }
 
+func (s *Service) UpdateIdentifierFirstLoginFlow(
+	ctx context.Context, flow string, body kClient.UpdateLoginFlowBody, cookies []*http.Cookie,
+) (*BrowserLocationChangeRequired, []*http.Cookie, error) {
+	ctx, span := s.tracer.Start(ctx, "kratos.Service.UpdateIdentifierFirstLoginFlow")
+	defer span.End()
+
+	csrfToken, ok := body.UpdateLoginFlowWithIdentifierFirstMethod.GetCsrfTokenOk()
+	if !ok {
+		return nil, nil, fmt.Errorf("missing csrf token")
+	}
+
+	identifier, ok := body.UpdateLoginFlowWithIdentifierFirstMethod.GetIdentifierOk()
+	if !ok {
+		return nil, nil, fmt.Errorf("missing identifier")
+	}
+
+	form := url.Values{}
+	form.Set("csrf_token", *csrfToken)
+	form.Set("identifier", *identifier)
+	form.Set("method", "identifier_first")
+
+	u, err := url.Parse(s.kratosPublicURL + "/self-service/login")
+	if err != nil {
+		return nil, nil, fmt.Errorf("invalid kratos URL: %w", err)
+	}
+	q := u.Query()
+	q.Set("flow", flow)
+	u.RawQuery = q.Encode()
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, u.String(), strings.NewReader(form.Encode()))
+	if err != nil {
+		s.logger.Errorf("failed to create http request: %s", err)
+		return nil, nil, err
+	}
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	for _, c := range cookies {
+		req.AddCookie(c)
+	}
```

Comment:
please isolate this "request building" logic in a separate private function

in alternative, you could unify the request building logic with the request invocation that happens at line 441

Reply:
put another comment on this since it has not been addressed yet, closing this in favor of the other comment